spec/grover/processor_spec.rb

```diff
@@ -581,5 +581,22 @@ def mean_colour_statistics(image)
         colours.map { |colour| image.data.dig('channelStatistics', colour, 'mean').to_s }
       end
     end
+
+    context 'when rendering HTML' do
+      let(:method) { :content }
+
+      let(:url_or_html) do
+        <<-HTML
+          <html>
+            <head></head>
+            <body>
+              <h1>Hey there</h1>
+            </body>
+          </html>
+        HTML
+      end
+
+      it { expect(convert.gsub(/[[:space:]]+/, '')).to eq url_or_html.gsub(/[[:space:]]+/, '') }
```

Comment:
Two things here.. I'd suggest using `Grover::Utils.squish` rather than removing all spaces. Would still give a consistent result whilst being true to the rendered output. Also, it feels like this would be a good opportunity to test that the *rendered* HTML is returned. ie something like: 
```
    context 'when rendering HTML' do
      let(:method) { :content }

      let(:url_or_html) do
        <<-HTML
          <html>
            <head></head>
            <body>
              <h1>Hey there</h1>
              <h2>Konnichiwa</h2>
              <script>document.querySelector('h1').remove();</script>
            </body>
          </html>
        HTML
      end

      it 'returns the rendered HTML' do
        expect(Grover::Utils.squish(convert)).to eq Grover::Utils.squish(<<~HTML)
          <html><head></head>
            <body>
              <h2>Konnichiwa</h2>
              <script>document.querySelector('h1').remove();</script>
            </body></html>
        HTML
      end
    end
```
Ie not just what you gave it, but rather the result of it being rendered.

Reply:
Good catch and point, thanks. Updated 🙌